File: client_leger/lib/classes/game-replay-commands/guess_difference_game_replay_command.dart
import 'package:client_leger/classes/game-replay-commands/game_replay_command.dart';
import 'package:client_leger/interfaces/dto/replay_event.dto.dart';
import 'package:client_leger/services/replay_service.dart';

class GuessDifferenceGameReplayCommand extends GameReplayCommand {
  final GuessDifferenceReplayDto eventDto;

  GuessDifferenceGameReplayCommand(
      {required this.eventDto, required super.time});

  @override
  void action() {
    ReplayService.to.handleClickResponse(
        eventDto.response, eventDto.body.imageArea, eventDto.body.coordinate);
  }
}

File: client/src/app/classes/guess-difference-game-replay-command/guess-difference-game-replay-command.ts
import { GameReplayCommand } from '@app/interfaces/game-replay-command';
import { ClassicModeService } from '@app/services/classic-mode/classic-mode.service';
import { ReplayEventDto } from '@common/model/events/replay.events';
import { Guess } from '@common/model/guess';
import { GuessResultClassic } from '@common/model/guess-result';

export class GuessDifferenceGameReplayCommand implements GameReplayCommand {
    time: number;
    private classicModeService: ClassicModeService;
    private replayEventDto: ReplayEventDto;

    constructor(classicModeService: ClassicModeService, replayEventDto: ReplayEventDto) {
        this.time = replayEventDto.time;
        this.classicModeService = classicModeService;
        this.replayEventDto = replayEventDto;
    }

    action() {
        this.classicModeService.handleClickCallback(this.replayEventDto.body as Guess, this.replayEventDto.response as GuessResultClassic);
    }
}
